api/v1alpha3/vspherecluster_types.go

```diff
@@ -39,6 +39,11 @@ type VSphereClusterSpec struct {
 	// +optional
 	Insecure *bool `json:"insecure,omitempty"`
 
+	// Thumbprint is the thumbprint of the given Server
+	// when supplied Insecure should not be set to true
+	// +optional
+	Thumbprint string `json:"thumbprint,omitempty"`
```

Comment:
Question:
Given that now `insecure==true` is equivalent to `Thumbprint == ""`, is it required to preserve the `insecure` flag? If not removing, at least what about deprecating?

Reply:
`insecure` is planned for deprecation as per our v1alpha4 roadmap. I'm planning on sending a different PR for this